velox/functions/sparksql/Arithmetic.h

```diff
@@ -368,6 +407,30 @@ FOLLY_ALWAYS_INLINE static int8_t fromHex(char c) {
 }
 } // namespace detail
 
+template <typename T>
+struct WidthBucketFunction {
+  template <typename TInput>
+  FOLLY_ALWAYS_INLINE bool call(
+      int64_t& result,
+      TInput value,
+      TInput min,
+      TInput max,
+      int64_t numBucket) {
```

Comment:
If only double is needed, could we drop the template parameter TInput?

Reply:
maybe in future spark would also support other data type, i thought keep current implement is ok since it is easy to extend for other data type. Or i can also drop the tempalte TInput if this option is recommand from code perspective?

What's the perfer option @rui-mo

Reply:
I'm open for it. Maybe let's follow up the community's suggestion.

Reply:
@Yuhta / @mbasmanova could you help review for this PR. thanks

Reply:
help, can anyone help confirm if this PR can be merged ? @rui-mo / @Yuhta / @mbasmanova